### PR TITLE
Fix empty inverse warping field

### PR DESCRIPTION
**Pull request by SaraDupont**
- Fix issue #1068 
- add comments and minor changes

### CODE REVIEW

File: scripts/sct_register_graymatter.py [hunk 1]
```diff
@@ -65,39 +66,49 @@ def register(self):
...
+            convert(self.fname_warp_target2template, tmp_dir+file_warp_target2template+ext_warp_target2template, squeeze_data=0)
 
         os.chdir(tmp_dir)
-        # TODO assert RPI, if not, change orientation
+        # save images
         im_automatic_ml.save()
         im_template_ml.save()
-        
+
+        # assert data is in RPI, if not change it to RPI
```
Comment (jcohenadad) [inline scripts/sct_register_graymatter.py]: is it still required, given that sct_register_multimodal (called later in the code) can now handle non-RPI data?

Comment (SaraDupont) [inline scripts/sct_register_graymatter.py]: Good point, I can remove it

**Comment by jcohenadad** [comment]
could you show the default value for the flag "-param <str>"

**Comment by jcohenadad** [comment]
default flag displays like:
~~~
 -param <str>                 Parameters for the multimodal registration between multilabel images 
                              Default value = 
                              step=1,type=im,a
                              lgo=slicereg,met
                              ric=MeanSquares:
                              step=2,type=im,a
                              lgo=syn,metric=M
                              eanSquares,iter=
                              10,smooth=0,shri
                              nk=2:step=3,type
                              =im,algo=bsplinesyn,metric=MeanSquares,iter=5,smooth=0
~~~
possible to make it nicer?

**Comment by SaraDupont** [comment]
I saw, but that would mean changes in the parser, which affects the display of all functions...

**Comment by jcohenadad** [comment]
OK, @peristeri could you please add a note regarding the parsing issue

**Comment by jcohenadad** [comment]
trying to optimize the parameters, i found something strange: 

when running register_multimodal using the temporary files, I obtain better results than when using the entire register_graymatter function with the same parameters. See below:

~~~
sct_register_multimodal -i multilabel_template_seg_crop.nii.gz -d multilabel_automatic_seg_crop.nii.gz -param step=1,type=seg,algo=centermassrot,smooth=1:step=2,type=im,algo=syn,iter=50,smooth=0,gradStep=1,metric=MeanSquares,slicewise=1 -iseg multilabel_template_seg_crop_bin.nii.gz -dseg multilabel_automatic_seg_crop_bin.nii.gz -r 0
~~~
output_7bc7yw

~~~
 sct_register_graymatter -gm crop_Smith_229826_sWIP_mFFE_065_SENSE_4_3_mask_GM.nii -wm crop_Smith_229826_sWIP_mFFE_065_SENSE_4_3_mask_WM.nii -w warp_template2anat.nii.gz -winv warp_anat2template.nii.gz -param step=1,type=seg,algo=centermassrot,smooth=1:step=2,type=im,algo=syn,iter=50,smooth=0,gradStep=1,metric=MeanSquares,slicewise=1
~~~
output_s9k8kq

**Comment by peristeri** [comment]
OK, @jcohenadad I'll look into it.

**Comment by SaraDupont** [comment]
How did you obtain the second screen shot ?
The warping field outputted by `sct_register_graymatter` is the concatenation of `warp_template2anat` and `warp_multilabel_template2auto` so it might make sense that you don't have the same result.

**Comment by peristeri** [comment]
The display error is cause by an edge case. In the function `msct_parser.DocSourceForge.align` the length of the default_value caused it to break it. The workaround is to increase the length to 120.

**Pull request merged**